File: 이솔/프로그래머스/Level2/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    arr=list(map(str,skill))
    for i in skill_trees:
        words=list(map(str,i))
        n=1
        answer+=1
        for j in words:
            if j not in arr:
              continue
            elif j in arr[:n]:
                n+=1            
            else:
                answer-=1
                break

    return answer

File: 문민우/프로그래머스/level2/스킬트리.java
package level2;

import java.util.Queue;
import java.util.LinkedList;

public class 스킬트리 {
    public int solution(String skill, String[] skill_trees) {

        int answer = 0;

        Queue<String> queue = new LinkedList<>();

        for (String skillTree : skill_trees) {

            for (int i = 0; i < skill.length(); i++) {
                queue.offer(Character.toString(skill.charAt(i)));
            }

            boolean flag = true;

            for (int i = 0; i < skillTree.length(); i++) {

                String cur = Character.toString(skillTree.charAt(i));

                if (skill.contains(cur)) {

                    if (queue.peek().equals(cur))
                        queue.poll();

                    else {

                        flag = false;
                        break;
                    }
                }
            }

            if (flag)
                answer++;

            queue.clear();
        }

        return answer;
    }
}
